src/resmom/linux/alps.c

```diff
@@ -3249,12 +3261,14 @@ inventory_to_vnodes(basil_response_t *brp)
 				NULL) == -1)
 				goto bad_vnl;
 
-			attr = "resources_available.PBScrayseg";
-			sprintf(utilBuffer, "%d", seg->ordinal);
-			if (vn_addvnr(nv, vname, attr, utilBuffer,
-				ATR_TYPE_STR, atype,
-				NULL) == -1)
-				goto bad_vnl;
+			if (vnode_per_numa_node) {
+				attr = "resources_available.PBScrayseg";
+				snprintf(utilBuffer, sizeof(utilBuffer), "%d",
```

Comment:
The way we have been copying data to this utility buf is inconsistent in this function. We should use snprintf all throughout.